tools/bag_processing/scripts/rosbag_fix_all.py

```diff
@@ -37,31 +37,90 @@ def dosys(cmd):
     return ret
 
 
-def rosbag_fix_all(inbag_paths_in, jobs, deserialize=False):
+def rosbag_fix_all(
+    inbag_paths_in, jobs, debayer, decode_haz, filter_topics, deserialize=False
+):
     this_folder = os.path.dirname(os.path.realpath(__file__))
     makefile = os.path.join(this_folder, "Makefile.rosbag_fix_all")
-    inbag_paths = [p for p in inbag_paths_in if not ".fix_all" in p]
+    inbag_paths_in = [p for p in inbag_paths_in if p.endswith(".bag")]
+
+    # Skip processed bags to start in the correct stage in the Makefile
+    inbag_paths = [p for p in inbag_paths_in if not ".rewrite_types.bag" in p]
+    inbag_paths = [p for p in inbag_paths if not ".migrate_old.bag" in p]
+    inbag_paths = [p for p in inbag_paths if not ".debayer.bag" in p]
+    inbag_paths = [p for p in inbag_paths if not ".depth_split.bag" in p]
+    inbag_paths = [p for p in inbag_paths if not ".fix_all.bag" in p]
+
     skip_count = len(inbag_paths_in) - len(inbag_paths)
     if skip_count:
         logging.info(
-            "Not trying to fix %d files that already end in .fix_all.bag", skip_count
+            "Not trying to fix %d files that already starts with fix_all_", skip_count
         )
+
     outbag_paths = [os.path.splitext(p)[0] + ".fix_all.bag" for p in inbag_paths]
     outbag_paths_str = " ".join(outbag_paths)
 
+    # Rosbag migrate message definitions
+    rewrite_types_args = "-v"
+
+    # Rosbag verify bags
     rosbag_verify_args = "-v"
     if deserialize:
         rosbag_verify_args += " -d"
 
+    # Rosbag convert debayer
+    if debayer:
+        rosbag_debayer_args = "-s"
+    else:
+        rosbag_debayer_args = "-n"
+
+    # Rosbag decode haz cam
+    if decode_haz is "":
+        rosbag_pico_split_extended_args = "-n"
+    else:
+        output_stream = os.popen("catkin_find --first-only bag_processing resources")
+        coeff_path = (
+            output_stream.read().rstrip() + "/" + decode_haz + "_haz_xyz_coeff.npy"
+        )
+        rosbag_pico_split_extended_args = "-s --in_npy " + coeff_path
```

Comment:
Might be good to assert here that `coeff_path` exists. If it doesn't, it will be much more confusing to track down the problem later.